### PR TITLE
[merged] dolphin-emu: 4.0.2 -> 5.0

### PR DESCRIPTION
**Pull request by jb55**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/emulators/dolphin-emu/default.nix [hunk 1]
```diff
@@ -1,13 +1,17 @@
-{ stdenv, pkgconfig, cmake, bluez, ffmpeg, libao, mesa, gtk2, glib
-, gettext, libpthreadstubs, libXrandr, libXext, readline
-, openal, libXdmcp, portaudio, SDL, wxGTK30, fetchurl
+{ stdenv, gcc5, pkgconfig, cmake, bluez, ffmpeg, libao, mesa, gtk2, glib
```
Comment (matthewbauer) [inline pkgs/misc/emulators/dolphin-emu/default.nix]: It's preferred to not have the compiler be a dependency. Instead, you should override "stdenv" in all-packages.nix

https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L6635

Comment (matthewbauer) [inline pkgs/misc/emulators/dolphin-emu/default.nix]: Also, gcc5 is the default library, so I think you can just leave it off.

File: pkgs/misc/emulators/dolphin-emu/default.nix [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+, gettext, git, libpthreadstubs, libXrandr, libXext, readline
```
Comment (matthewbauer) [inline pkgs/misc/emulators/dolphin-emu/default.nix]: "git" can be removed. fetchFromGitHub should be all you need.

File: pkgs/misc/emulators/dolphin-emu/default.nix [hunk 3]
```diff
@@ -20,16 +24,17 @@ stdenv.mkDerivation rec {
...
+                  libevdev git libXdmcp portaudio libusb libpulseaudio ];
 
   meta = {
     homepage = http://dolphin-emu.org/;
     description = "Gamecube/Wii/Triforce emulator for x86_64 and ARM";
     license = stdenv.lib.licenses.gpl2;
-    platforms = stdenv.lib.platforms.linux;
     maintainers = with stdenv.lib.maintainers; [ MP2E ];
-    broken = true;
+    # x86_32 is an unsupported platform.
```
Comment (matthewbauer) [inline pkgs/misc/emulators/dolphin-emu/default.nix]: We may want to archive the old version as "4.0.nix" so we can still have this available for 32-bit.

File: pkgs/misc/emulators/dolphin-emu/default.nix (hunk 2 above)
Comment (jb55) [inline pkgs/misc/emulators/dolphin-emu/default.nix]: /cc @MP2E on this one, I copied this over from `master.nix` so I figured it was there for a reason...

File: pkgs/misc/emulators/dolphin-emu/default.nix (hunk 1 above)
Comment (jb55) [inline pkgs/misc/emulators/dolphin-emu/default.nix]: /cc @MP2E

**Comment by MP2E** [comment]
Hey there! Sorry for the late response. Been meaning to get more active again

There was a period where gcc 4.8 or 4.9 was the default and dolphin-emu needed gcc5, so gcc5 can safely be removed.

As for git, it was put in for the nightly development builds because dolphin calls the git executable during configuration. I'm sure it can safely be removed from this. I'll update my NixOS install tonight and give this a test!

**Comment by MP2E** [comment]
Merged as of 7129ebbd5e107be149d4af09f3da90094ec50b14, thank you for contributing!

**Pull request closed without merging**